nemo/collections/nlp/models/text_classification/text_classification_model.py

```diff
@@ -46,13 +46,17 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         """Initializes the BERTTextClassifier model.
         """
 
+        # Q: Why there is a separate entity, stored and managed independently of self._cfg
         # shared params for dataset and data loaders
         self.dataset_cfg = cfg.dataset
 
+        # Q: why tokenizer is initialized before init of parent class?
```

Comment:
The super class constructor calls setup_training_data, which uses tokenizer to prepare the data. That is why tokenizer needs to be initialized first before super constructor